src/main/java/seedu/address/model/person/Picture.java

```diff
@@ -19,29 +20,42 @@
     // 10 MB
     public static final long MAX_FILE_SIZE = 10 * 1024 * 1024;
 
+    public static final Set<Byte[]> IMAGE_MAGIC_NUMBERS = new HashSet<>();
+
+    // Initialize magic numbers of images. null represents any value for 2 bytes.
+    static {
+        // png
+        IMAGE_MAGIC_NUMBERS.add(FileUtil.intArrayToByteArray(new int[] {0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A,
+                0x0A}));
+        // jpg, jpeg
+        IMAGE_MAGIC_NUMBERS.add(FileUtil.intArrayToByteArray(new int[] {0xFF, 0xD8, 0xFF, 0xDB}));
+        IMAGE_MAGIC_NUMBERS.add(FileUtil.intArrayToByteArray(new int[] {0xFF, 0xD8, 0xFF, 0xE0, 0x00, 0x10, 0x4A, 0x46,
+                0x49, 0x46, 0x00, 0x01}));
+        IMAGE_MAGIC_NUMBERS.add(FileUtil.intArrayToByteArray(new int[] {0xFF, 0xD8, 0xFF, 0xEE}));
+        IMAGE_MAGIC_NUMBERS.add(FileUtil.intArrayToByteArray(new int[] {0xFF, 0xD8, 0xFF, 0xE1, -1, -1, 0x45, 0x78,
+                0x69, 0x66, 0x00, 0x00}));
+    }
+
     private final Path filePath;
 
     public Picture(Path filePath) {
         this.filePath = filePath;
     }
 
+    public static boolean isValidFilePath(Path path) {
```

Comment:
Can you add in the `isValidImage` check here also? I think this is check is used when deserialising from the json data file.